pkg/kubectl/resource/mapper.go

```diff
@@ -78,14 +84,14 @@ func (m *Mapper) InfoForObject(obj runtime.Object, typer runtime.ObjectTyper, pr
 		return nil, fmt.Errorf("unable to get type info from the object %q: %v", reflect.TypeOf(obj), err)
 	}
 
-	groupVersionKind := groupVersionKinds[0]
+	gvk := groupVersionKinds[0]
 	if len(groupVersionKinds) > 1 && len(preferredGVKs) > 0 {
-		groupVersionKind = preferredObjectKind(groupVersionKinds, preferredGVKs)
+		gvk = preferredObjectKind(groupVersionKinds, preferredGVKs)
 	}
 
-	mapping, err := m.RESTMapper.RESTMapping(groupVersionKind.GroupKind(), groupVersionKind.Version)
+	mapping, err := m.RESTMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
```

Comment:
shouldn't this have been deleted if it was moved into the `if m.localFn == nil || !m.localFn()` block below? same question for the `client, err := m.ClientMapper.ClientForMapping(mapping)` bit

Reply:
> shouldn't this have been deleted if it was moved into the if m.localFn == nil || !m.localFn() block below? same question for the client, err := m.ClientMapper.ClientForMapping(mapping) bit

fixed